pycodestyle.py

```diff
@@ -2056,6 +2056,25 @@ def __init__(self, *args, **kwargs):
             if 'paths' in options_dict:
                 self.paths = options_dict['paths']
 
+        # After options.__dict__.update we need to ensure the *args and **kwargs where correctly passed
+        assert isinstance(options.verbose, int)
```

Comment:
This disappear with `python -O` so I don't think this does what you want.

Reply:
I just ran `python3 -O pycodestyle.py --testsuite testsuite` and `verbose` option was present on the debugging statements I had added:
```Python
StyleGuide: args: ()
StyleGuide: kwargs: {'parse_argv': True}
StyleGuide: process_options, select: []
StyleGuide: process_options, ignore: []
StyleGuide: process_options, select: []
StyleGuide: process_options, ignore: []
StyleGuide: options: {'show_source': False, 'statistics': False, 'select': [], 'quiet': 0, 'filename': ['*.py'], 'count': False, 'benchmark': False, 'repeat': True, 'testsuite': 'testsuite', 'reporter': None, 'exclude': ['.svn', 'CVS', '.bzr', '.hg', '.git', '__pycache__', '.tox'], 'verbose': 0, 'format': 'default', 'ignore': [], 'show_pep8': False, 'diff': False, 'max_line_length': 79, 'hang_closing': False, 'config': False, 'doctest': False}
StyleGuide: options.ignore 1: []
StyleGuide: options.ignore 2: ()
3061 lines tested: 37 files, 381 test cases.
Test passed.
```

But I already had created this function:
```python
File: pycodestyle/pycodestyle.py
2461: def customisinstance(values_object, option_name, target_type):
2462:     """ Used to check optional options like verbose which is not present
2463:     when running with `python -o`"""
2464:     values_dict = values_object.__dict__
2465:     if option_name in values_dict:
2466:         return isinstance(values_dict[option_name], target_type)
2467:     return True
```